php/stats.php

```diff
@@ -86,52 +92,108 @@
+)
+UNION 
+(
+    SELECT '%s', ROUND(f.distance %s) AS distance, DATE_FORMAT(duration, '%%H:%%i') AS duration, s.iata, s.icao, s.apid, d.iata, d.icao, d.apid
+    FROM flights AS f, airports AS s, airports AS d
+    WHERE f.src_apid = s.apid AND f.dst_apid = d.apid AND $filter
+    ORDER BY distance ASC
+    LIMIT 1
+)
+SQL,
     _("Longest"),
     $multiplier,
     _("Shortest"),
     $multiplier
 );
 $sth = $dbh->query($sql);
-$first = true;
+$rows = [];
 foreach ($sth as $row) {
-    if ($first) {
-        $first = false;
-    } else {
-        printf(";");
-    }
     $src_code = format_apcode2($row[3], $row[4]);
     $dst_code = format_apcode2($row[6], $row[7]);
-    printf("%s,%s %s,%s,%s,%s,%s,%s", $row[0], $row[1], $unit, $row[2], $src_code, $row[5], $dst_code, $row[8]);
+    $rows[] = sprintf(
+        "%s,%s %s,%s,%s,%s,%s,%s",
+        $row[0],
+        $row[1],
+        $unit,
+        $row[2],
+        $src_code,
+        $row[5],
+        $dst_code,
+        $row[8]
+    );
 }
-printf("\n");
+echo implode("\t", $rows) . "\n";
 
 // North, South, West, East
 // 0 desc, 1 iata, 2 icao, 3 apid, 4 x, 5 y
 $sql = sprintf(
-    "(SELECT '%s',iata,icao,apid,x,y FROM airports WHERE y=(SELECT MAX(y) FROM airports AS a, flights AS f WHERE (f.src_apid=a.apid OR f.dst_apid=a.apid) AND " . $filter . ") ORDER BY iata LIMIT 1) UNION " .
-    "(SELECT '%s',iata,icao,apid,x,y FROM airports WHERE y=(SELECT MIN(y) FROM airports AS a, flights AS f WHERE (f.src_apid=a.apid OR f.dst_apid=a.apid) AND " . $filter . ") ORDER BY iata LIMIT 1) UNION " .
-    "(SELECT '%s',iata,icao,apid,x,y FROM airports WHERE x=(SELECT MIN(x) FROM airports AS a, flights AS f WHERE (f.src_apid=a.apid OR f.dst_apid=a.apid) AND " . $filter . ") ORDER BY iata LIMIT 1) UNION " .
-    "(SELECT '%s',iata,icao,apid,x,y FROM airports WHERE x=(SELECT MAX(x) FROM airports AS a, flights AS f WHERE (f.src_apid=a.apid OR f.dst_apid=a.apid) AND " . $filter . ") ORDER BY iata LIMIT 1)",
+<<<SQL
+(
+    SELECT '%s', iata, icao, apid, x, y
+    FROM airports WHERE y = (
+        SELECT MAX(y)
+        FROM airports AS a, flights AS f
+        WHERE (f.src_apid = a.apid OR f.dst_apid = a.apid) AND $filter
+    )
+    ORDER BY iata
+    LIMIT 1
+)
+UNION
+(
+    SELECT '%s',iata,icao,apid,x,y
+    FROM airports
+    WHERE y = (
+        SELECT MIN(y)
+        FROM airports AS a, flights AS f
+        WHERE (f.src_apid = a.apid OR f.dst_apid = a.apid) AND $filter
+    )
+    ORDER BY iata
+    LIMIT 1
+)
+UNION
+(
+    SELECT '%s',iata,icao,apid,x,y
+    FROM airports
+    WHERE x = (
+        SELECT MIN(x)
+        FROM airports AS a, flights AS f
+        WHERE (f.src_apid = a.apid OR f.dst_apid = a.apid) AND $filter
+    )
+    ORDER BY iata
+    LIMIT 1
+)
+UNION
+(
+    SELECT '%s',iata,icao,apid,x,y
+    FROM airports WHERE x=(
+        SELECT MAX(x) FROM airports AS a, flights AS f WHERE (f.src_apid=a.apid OR f.dst_apid=a.apid) AND $filter
+    )
+    ORDER BY iata
+    LIMIT 1
```

Comment:
This part is slightly inconsistent but I've already rewritten this whole query to fix #993.